src/Uno.UWP/Devices/Sensors/Compass.wasm.cs

```diff
@@ -0,0 +1,148 @@
+#nullable enable
+
+using System;
+using System.Numerics;
+
+namespace Windows.Devices.Sensors;
+
+public partial class Compass
+{
+	private Accelerometer? _accelerometer;
+	private Magnetometer? _magnetometer;
+
+	private Vector3? _lastAccelerometer;
+	private Vector3? _lastMagnetometer;
+
+	private uint _reportInterval;
+
+	/// <summary>
+	/// Gets or sets the current report interval for the compass.
+	/// </summary>
+	public uint ReportInterval
+	{
+		get => _reportInterval;
+		set
+		{
+			if (_reportInterval == value)
+			{
+				return;
+			}
+
+			lock (_syncLock)
+			{
+				_reportInterval = value;
+
+				if (_accelerometer == null || _magnetometer == null)
+				{
+					return;
+				}
+
+				if (_readingChanged != null)
+				{
+					//restart reading to apply interval
+					StopReadingChanged();
+					StartReadingChanged();
+				}
+			}
+		}
+	}
+
+	private static Compass? TryCreateInstance()
+	{
+		var accelerometer = Accelerometer.GetDefault();
+		var magnetometer = Magnetometer.GetDefault();
+
+		if (accelerometer != null && magnetometer != null)
+		{
+			return new Compass();
+		}
+
+		return null;
+	}
+
+	private void StartReadingChanged()
+	{
+		_accelerometer = Accelerometer.GetDefault();
+		_magnetometer = Magnetometer.GetDefault();
+
+		if (_accelerometer != null && _magnetometer != null)
```

Comment:
We won't subscribe to ReadingChanged for the accelerometer if the magnetometer is null? Or vice-versa?

Reply:
Oh nevermind, I see :)